internal/pkg/cli/pipeline_deploy.go

```diff
@@ -123,10 +128,13 @@ func (o *deployPipelineOpts) Execute() error {
 	if err != nil {
 		return fmt.Errorf("unmarshal pipeline manifest: %w", err)
 	}
-	if len(pipeline.Name) > 100 {
-		return fmt.Errorf(`pipeline name '%s' must be shorter than 100 characters`, pipeline.Name)
+	// Set pipeline name if not passed in with flag.
+	if o.name == "" {
+		if len(pipeline.Name) > 100 {
+			return fmt.Errorf(`pipeline name '%s' must be shorter than 100 characters`, pipeline.Name)
+		}
```

Comment:
What do you think of moving this logic to `manifest/validate.go`?
```go
func (m PipelineManifest) Validate() error 
```
That could be a nice way of having a similar interface between workload manifests and pipeline manifests.
And then we can remove `if o.name == ""` replacing it with:
```go
if err := pipeline.Validate(); err != nil { ... }
```